x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/plan/AbstractNodeSerializationTests.java

```diff
@@ -57,7 +59,12 @@ protected T copyInstance(T instance, TransportVersion version) throws IOExceptio
             getNamedWriteableRegistry(),
             (out, v) -> new PlanStreamOutput(out, configuration()).writeNamedWriteable(v),
             in -> {
-                PlanStreamInput pin = new PlanStreamInput(in, in.namedWriteableRegistry(), configuration());
+                PlanStreamInput pin = new PlanStreamInput(
+                    in,
+                    in.namedWriteableRegistry(),
+                    configuration(),
+                    new SerializationTestUtils.TestNameIdMapper()
```

Comment:
This ensures we correctly check the name ids in serialization tests, but don't have to patch them up in every equality check; instead, we deserialize ids to the original ones in tests.